src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -50,6 +50,12 @@ public static string HtmlEncode(string value)
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        public static void HtmlEncode(string value, TextWriter output)
+        {
+            // Since we already have a private method that decodes the value using a StringBuilder we don't want to have duplicate code in order to have a better mantainance.
+            output.Write(HtmlEncode(value));
```

Comment:
@safern I think they are just as good. Splitting into base classes makes it easier if you have more than 2 methods you would need to wrap. Otherwise you end up with a lot of lambdas you need to write :). 

There should not be a lot of performance difference between calling via a delegate and calling via callvirt. 

And if we don't think the original code is used on the hot path, then this becomes a question of what is the nicest way to write the code and @justinvp came up with an elegant one.